Common/src/gRPC/Validators/EventSubscriptionRequestValidator.cs

```diff
@@ -0,0 +1,124 @@
+// This file is part of the ArmoniK project
+// 
+// Copyright (C) ANEO, 2021-2023. All rights reserved.
+// 
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+// 
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY, without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+// 
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+
+using ArmoniK.Api.gRPC.V1.Events;
+using ArmoniK.Api.gRPC.V1.Results;
+using ArmoniK.Api.gRPC.V1.Tasks;
+
+using FluentValidation;
+using FluentValidation.Results;
+
+namespace ArmoniK.Core.Common.gRPC.Validators;
+
+/// <summary>
+///   Validator for <see cref="EventSubscriptionRequest" />
+/// </summary>
+public class EventSubscriptionRequestValidator : AbstractValidator<EventSubscriptionRequest>
+{
+  /// <inheritdoc />
+  public override ValidationResult Validate(ValidationContext<EventSubscriptionRequest> context)
+  {
+    if (context.InstanceToValidate.ResultsFilters != null)
+    {
+      foreach (var filtersAnd in context.InstanceToValidate.ResultsFilters.Or)
+      {
+        foreach (var filterField in filtersAnd.And)
+        {
+          switch (filterField.Field.FieldCase)
+          {
+            case ResultField.FieldOneofCase.ResultRawField:
+              switch (filterField.Field.ResultRawField.Field)
+              {
+                case ResultRawEnumField.Status:
+                case ResultRawEnumField.CreatedAt:
+                case ResultRawEnumField.CompletedAt:
+                case ResultRawEnumField.Unspecified:
+                  context.AddFailure($"Cannot filter {filterField.Field.ResultRawField.Field} on in this API");
+                  break;
+                case ResultRawEnumField.OwnerTaskId:
+                case ResultRawEnumField.SessionId:
+                case ResultRawEnumField.Name:
+                case ResultRawEnumField.ResultId:
+                  break;
+                default:
+                  throw new ArgumentOutOfRangeException();
```

Comment:
`throw` or `context.AddFailure`?

Reply:
At this point, it is not the request that is not valid but our code so I think the exception is more appropriate.

Reply:
I think this can happen if the client sends an invalid number for the enum (like 100).